### PR TITLE
Replace mkdirp with fs.mkdirSync(path, { recursive: true })

### LINKED ISSUE #227931: Replace `mkdirp` with `fs.mkdirSync(path, { recursive: true })`
The `mkdirp` utility is used at multiple places in vscode https://github.com/search?q=repo%3Amicrosoft%2Fvscode+mkdirp+path%3Apackage.json&type=code

It can be replaced with Node.js native fs.mkdirSync added `recursive` option in Node.js [v10.12.0](https://github.com/nodejs/node/releases/tag/v10.12.0) published on Oct 10, 2018.

The `mkdirp` currently uses these native APIs:
* https://github.com/isaacs/node-mkdirp/blob/fdf6005432c2506ce31f8cae237039dda2f8479d/src/opts-arg.ts#L4-L8
* https://github.com/isaacs/node-mkdirp/blob/fdf6005432c2506ce31f8cae237039dda2f8479d/src/mkdirp-native.ts#L13-L19

### PR DESCRIPTION
**Pull request by bpasero**
(fix #227931)

### CODE REVIEW

File: test/smoke/src/main.ts [hunk 1]
```diff
@@ -179,7 +178,7 @@ function parseQuality(): Quality {
 //
 if (!opts.web) {
 	let testCodePath = opts.build;
-	let electronPath: string;
+	let electronPath: string | undefined;
```
Comment (trivikr) [inline test/smoke/src/main.ts:181]: This change seems to be unrelated to removal of mkdirp
```suggestion
	let electronPath: string;
```

File: build/lib/builtInExtensions.ts [hunk 2]
```diff
@@ -15,8 +15,6 @@ import * as fancyLog from 'fancy-log';
 import * as ansiColors from 'ansi-colors';
```
Comment (trivikr) [inline build/lib/builtInExtensions.ts:15]: It appears like the transpiled file `build/lib/builtInExtensions.js` needs to be committed.

CI Failure: https://dev.azure.com/vscode/VSCode/_build/results?buildId=131114&view=logs&j=c7493abb-a1f4-533f-2d24-71780a69f247&t=8d02ba2f-f021-560e-034c-effb4dd4b969

File: test/smoke/src/main.ts (hunk 1 above)
Comment (bpasero) [reply test/smoke/src/main.ts:181]: thats ok 👍

File: build/lib/builtInExtensions.ts (hunk 2 above)
Comment (bpasero) [reply build/lib/builtInExtensions.ts:15]: Yes, thanks!

**Review by joyceerhl — APPROVED** [review]
